nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -15,18 +15,20 @@
 #ifndef NAV2_BEHAVIOR_TREE__BT_ACTION_SERVER_IMPL_HPP_
 #define NAV2_BEHAVIOR_TREE__BT_ACTION_SERVER_IMPL_HPP_
 
-#include <memory>
-#include <string>
+#include <chrono>
+#include <exception>
 #include <fstream>
+#include <limits>
+#include <memory>
 #include <set>
-#include <exception>
+#include <string>
 #include <vector>
-#include <limits>
 
 #include "nav2_msgs/action/navigate_to_pose.hpp"
 #include "nav2_behavior_tree/bt_action_server.hpp"
-#include "ament_index_cpp/get_package_share_directory.hpp"
 #include "nav2_util/node_utils.hpp"
+#include "rcl_action/action_server.h"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
```

Comment:
I'm curious how you identified the headers to remove or the C libs to add in in place of others (here and in other files)? I expect this was a bit trial and error as you were adding in the specific targets within each package, but if there was something more automated you used that would be good for me to have in my toolbox as a future reviewer of changes to this codebase after this PR

Reply:
Good question.

It was completely manual for right now.  For further work here, I'm going to look into doing something much more automated, because it's not particularly sustainable with the amount of code that exists in this repository.